win32ss/user/ntuser/menu.c

```diff
@@ -3282,10 +3282,14 @@ static void FASTCALL MENU_HideSubPopups(PWND pWndOwner, PMENU Menu,
       {
           PWND pWnd;
           if (!VerifyMenu(Item->spSubMenu)) return;
-          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
           MENU_HideSubPopups(pWndOwner, Item->spSubMenu, FALSE, wFlags);
           MENU_SelectItem(pWndOwner, Item->spSubMenu, NO_SELECTED_ITEM, SendMenuSelect, NULL);
           TRACE("M_HSP top p hm %p  pWndOwner IDMenu %p\n",top_popup_hmenu,pWndOwner->IDMenu);
+
+          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
+          if (!pWnd)
+             return;
+
           co_UserDestroyWindow(pWnd);
```

Comment:
I wonder whether one should also continue taking the code below? What happens if you do this instead:
```suggestion
          pWnd = ValidateHwndNoErr(Item->spSubMenu->hWnd);
          if (pWnd)
              co_UserDestroyWindow(pWnd);
```
? :thinking:

Reply:
I'm wrong here. The code below is still required, otherwise the previous windows will freeze.
Fixed.